### PR TITLE
Rework Mirror service

### PR DESCRIPTION
**Pull request by sahibamittal**
### Description

Mirror-service is being reworked in Hyades. Corresponding changes need to be done in apiserver:

- [x] Input topic for mirroring commands have changed; Previously multiple, now only one
- [x] Parsing has to be changed to Protobuf
- [x] Account for vulnerabilities from GitHub when parsing

### Addressed Issue

https://github.com/DependencyTrack/hyades/issues/451

### Checklist

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/github-templates/CONTRIBUTING.md#pull-requests)
- [ ] This PR fixes a defect, and I have provided tests to verify that the fix is effective
- [x] This PR implements an enhancement, and I have provided tests to verify that it works as intended
- [ ] This PR introduces changes to the database model, and I have added corresponding [update logic](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade)
- [ ] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly

### CODE REVIEW

File: src/main/java/org/dependencytrack/event/kafka/processor/MirrorVulnerabilityProcessor.java [hunk 1]
```diff
@@ -35,19 +37,14 @@ public void process(final Record<String, Bom> record) {
...
+            Vulnerability.Source source = Vulnerability.Source.valueOf(mirrorSource);
             final Vulnerability vulnerability = mapBomToVulnerability(qm, bom);
             final List<VulnerableSoftware> vsListOld = qm.detach(qm.getVulnerableSoftwareByVulnId(vulnerability.getSource(), vulnerability.getVulnId()));
             final Vulnerability synchronizedVulnerability = qm.synchronizeVulnerability(vulnerability, false);
-            org.cyclonedx.model.vulnerability.Vulnerability cycloneVuln = bom.getVulnerabilities().get(0);
-            if (cycloneVuln.getReferences() != null) {
-                for (int i = 0; i < cycloneVuln.getReferences().size(); i++) {
-                    final String alias = cycloneVuln.getReferences().get(i).getId();
+            var cycloneVuln = bom.getVulnerabilities(0);
+            if (cycloneVuln.getReferencesList() != null && !cycloneVuln.getReferencesList().isEmpty()) {
```
[the remark this answers is not in the document]
Comment (sahibamittal) [reply src/main/java/org/dependencytrack/event/kafka/processor/MirrorVulnerabilityProcessor.java:45]: Alright, will remove null checks and will keep size check.

**Review by nscuro — APPROVED** [review]

**Comment by sahibamittal** [comment]
**Merge only when PR in Hyades is merged for same.**

File: src/main/java/org/dependencytrack/event/kafka/KafkaEventDispatcher.java [hunk 2]
```diff
@@ -60,9 +60,9 @@ public Future<RecordMetadata> dispatchAsync(final Event event, final Callback ca
         } else if (event instanceof final ComponentRepositoryMetaAnalysisEvent e) {
             return dispatchAsyncInternal(KafkaEventConverter.convert(e), callback);
         } else if (event instanceof final OsvMirrorEvent e) {
-            return dispatchAsyncInternal(new KafkaEvent<>(KafkaTopics.MIRROR_OSV, e.ecosystem(), "", null), callback);
+            return dispatchAsyncInternal(new KafkaEvent<>(KafkaTopics.VULNERABILITY_MIRROR_COMMAND, e.ecosystem(), "", null), callback);
         } else if (event instanceof NistMirrorEvent) {
-            return dispatchAsyncInternal(new KafkaEvent<>(KafkaTopics.MIRROR_NVD, UUID.randomUUID().toString(), "", null), callback);
+            return dispatchAsyncInternal(new KafkaEvent<>(KafkaTopics.VULNERABILITY_MIRROR_COMMAND, UUID.randomUUID().toString(), "", null), callback);
```
Comment (nscuro) [inline src/main/java/org/dependencytrack/event/kafka/KafkaEventDispatcher.java:65]: The event key will need to be the data source name, and the value the list of ecosystems (for OSV), or `""` / `null` (for NVD and GITHUB). And I think we're missing the GitHub event?

Comment (sahibamittal) [reply src/main/java/org/dependencytrack/event/kafka/KafkaEventDispatcher.java:65]: Yes I deleted my change somehow while merging from main..

**Review by nscuro — APPROVED** [review]

**Pull request merged**